editor/editor_properties.cpp

```diff
@@ -477,42 +477,32 @@ EditorPropertyCheck::EditorPropertyCheck() {
 
 void EditorPropertyEnum::_option_selected(int p_which) {
 
-	String text = options->get_item_text(p_which);
-	Vector<String> text_split = text.split(":");
-	if (text_split.size() == 1) {
-		emit_signal("property_changed", get_edited_property(), p_which);
-		return;
-	}
-	String name = text_split[1];
-	emit_signal("property_changed", get_edited_property(), name.to_int());
+	int val = options->get_item_metadata(p_which);
+	emit_signal("property_changed", get_edited_property(), val);
 }
 
 void EditorPropertyEnum::update_property() {
 
 	int which = get_edited_object()->get(get_edited_property());
-	if (which == 0) {
-		options->select(which);
-		return;
-	}
```

Comment:
The removal of this condition fixes #23022

Imagine the following enum:

```
enum SomeEnum{
    EntryA = -2,
    EntryB, #-1 
    EntryC, #0
    EntryD #1
}
```

Selecting `EntryC` in the Inspector would mean selecting the value 0. The above code selects the first entry (the one at index 0: `EntryA`) if the value is 0, wrongly assuming the value of the first entry is always 0 (it's -2 in this case).